Convert dynamic targets 1 to static

padilla410 opened the pull request:
The PR converts the dynamic version of targets 1 to a static version. It does not resolve any of the issues @gcook-usgs and I are tracking [here](https://github.com/padilla410/ds-pipelines-targets-1-course-static/issues). I intend to migrate those issues over to the USGS-R repo once this PR has been merged.

I initially developed this static version on my personal github because there was a lot of uncertainty about the DGEC migration. I think Jesse, Althea, and I have worked through a lot of that, so I feel comfortable rolling these back into USGS-R and prepping them for migration.

Also did a few things to make the content in this repo "safer":
1. I archived the original dynamic version of the training as a release
2. I created an `archive` subfolder in the repo to house the images (in case MS does something to break that hosting service)
3. I archived a copy of the original template repo in `archive`

ceenell commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+We'll briefly explain some of the functions and conventions used here. For more extensive explanations, visit the [`targets` documentation](https://books.ropensci.org/targets/walkthrough.html). 
+
+  - As you would with normal R scripts, put any `source` commands for loading R files and `library` commands for loading packages at the top of the file. The packages loaded here should be only those needed to build the targets _plan_; packages needed to build specific _targets_ can be loaded later.
```
"As you would with normal R scripts" is making assumptions about the trainee and what is "normal"

padilla410 replied:
I do think this falls into the category of "normal" - esp. with a "lazy loading" language like R, however I see your point and have removed that assumption.

ceenell commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+We'll briefly explain some of the functions and conventions used here. For more extensive explanations, visit the [`targets` documentation](https://books.ropensci.org/targets/walkthrough.html). 
+
+  - As you would with normal R scripts, put any `source` commands for loading R files and `library` commands for loading packages at the top of the file. The packages loaded here should be only those needed to build the targets _plan_; packages needed to build specific _targets_ can be loaded later.
```
```suggestion
  - As you would with normal R scripts, put any `source` commands for loading R files and `library` commands for loading packages at the top of the file. The packages loaded in `tar_option_set()` should be only those needed to build the targets _plan_; packages needed to build specific _targets_ are loaded later.
```

ceenell commented on course-instructions.md:
````diff
@@ -0,0 +1,342 @@
+)
+```
+
+This file defines the relationships between different "targets" (see how the target `model_RMSEs_csv` is an input to the command that creates the target `eval_data`?), tells us where to find any functions that are used to build targets (see the `source` call that points you to `code.R`), and declares the package dependencies needed to build the different targets (see the `target_option_set()` com ...
````
Break this sentence up

padilla410 replied:
done!

ceenell commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+  - As you would with normal R scripts, put any `source` commands for loading R files and `library` commands for loading packages at the top of the file. The packages loaded here should be only those needed to build the targets _plan_; packages needed to build specific _targets_ can be loaded later.
+  - Declare each target by using the function `tar_target()` and passing in a target name (`name` arg) and the expression to run to build the target (`command` arg).
+  - There are two types of targets - objects and files. If your target is a file, you need to add `format = "file"` to your `tar_target()` call and the command needs to return the filename of the new file.
```
```suggestion
  - There are two types of targets - objects and files. If your target is a file, you need to add `format = "file"` to your `tar_target()` call and the command needs to return the filename of the new file. `figure_1_png` above is a file target.
```

ceenell commented on course-instructions.md:
````diff
@@ -0,0 +1,342 @@
+)
+```    
+
+then use `Ctrl+D` to exit the file creation mode and return to the prompt. 
````
This is fine, but curious as to why this is done in terminal

padilla410 replied:
This is a hold over from the dynamic training and was probably part of the original scipiper training. I think it is probably a verrry gentle introduction to using `vim` to do stuff.

ceenell commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+<details>
+<summary><h2>Modifying Your Pipeline by Adding to Your Pull Request</h2></summary>
+
+* Great work, but it would be great to download the `model_RMSEs.csv` to `1_fetch/out/model_RMSEs.csv` instead. Can you make a change to make sure that happens when you build this target?
```
```suggestion
* Great work, but now you've realized you want to download the `model_RMSEs.csv` to `1_fetch/out/model_RMSEs.csv` instead. Can you make a change to make sure that happens when you build this target?
```

ceenell commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+---
+
+Now - adding on to this PR by pushing up additional commits - add and modify code, folders, and this makefile so you can build _your_ `3_visualize/out/figure_1.png` with `targets::tar_make()` (BTW, if you haven't run into this syntax, {package_name}::{function_name} allows you to run a function without `library({package_name})`). Build on your existing folders, functions, and code from the previo ...
```
First sentence unclear

padilla410 replied:
Added a commit to clarify language here

ceenell commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+Now - adding on to this PR by pushing up additional commits - add and modify code, folders, and this makefile so you can build _your_ `3_visualize/out/figure_1.png` with `targets::tar_make()` (BTW, if you haven't run into this syntax, {package_name}::{function_name} allows you to run a function without `library({package_name})`). Build on your existing folders, functions, and code from the previo ...
+
+There is a file in your repo called `.gitignore`. Add a line with `_targets` so that git doesn't track the contents of the `targets` directory. In addition, make sure that the contents of `1_fetch/out/*` will be ignored by git because you don't want to store changes to your downloaded .csv files (you may have already added a more generic `*/out/*` to your `.gitignore` that handles all `out/` fold ...
```
Wonder if this should be like "Whoops! Did you commit your local `_targets` folder? This folder should be ignored..."

ceenell replied:
Often I am seeing people committing this first, and then not being able to uncommit it when they realize it should be ignored

padilla410 replied:
Yes, It think most people make a mistake here. `gitignore` first gets mentioned at [line 125](https://github.com/padilla410/ds-pipelines-targets-1-course-static/blob/main/course-instructions.md?plain=1#L125). I think it ultimately happens because [line 298](https://github.com/padilla410/ds-pipelines-targets-1-course-static/blob/b4133804c68e11abcf2cbd02482fff0cc6452f96/course-instructions.md?plain=1#L298) says "add and modify code, folders, and this makefile so you can build _your_ `3_visualize/out/figure_1.png` with `targets::tar_make()`".

I think the time to address the `_targets` subfolder is right after the first `tar_make` command.

padilla410 replied:
modified

ceenell commented on my_work_R/my_happy_script.R:
```diff
@@ -0,0 +1,94 @@
+
+
+
+png(file = file.path(project_output_dir, 'figure_1.png'), width = 8, height = 10, res = 200, units = 'in')
+par(omi = c(0,0,0.05,0.05), mai = c(1,1,0,0), las = 1, mgp = c(2,.5,0), cex = 1.5)
+
+plot(NA, NA, xlim = c(2, 1000), ylim = c(4.7, 0.75),
+     ylab = "Test RMSE (°C)", xlab = "Training temperature profiles (#)", log = 'x', axes = FALSE)
+
+n_profs <- c(2, 10, 50, 100, 500, 980)
+
+axis(1, at = c(-100, n_profs, 1e10), labels = c("", n_profs, ""), tck = -0.01)
+axis(2, at = seq(0,10), las = 1, tck = -0.01)
+
+# slight horizontal offsets so the markers don't overlap:
+offsets <- data.frame(pgdl = c(0.15, 0.5, 3, 7, 20, 30)) %>%
+  mutate(dl = -pgdl, pb = 0, n_prof = n_profs)
```
We should really convert this to ggplot2.

padilla410 replied:
Noted in #20

ceenell reviewed (changes requested):
I've provided a number of text edits to make the progression clearer. In general, the text feels under linked. I think a few more resources could be pointed to. As much as possible I tried to provide text edits that you can accept. These were intended to use reduce ambiguity.

There are several spots that would benefit from a single graphic or screenshot - directory structure in particular. IMO we also need to update that plot to use ggplot, but can save that for later to move things forward right now.

padilla410 commented on course-instructions.md:
````diff
@@ -0,0 +1,342 @@
+install.packages(c("tidyverse", "sbtools", "whisker", "dataRetrieval"))
+```
+
+When all packages are installed, check the version of the `targets` package with `packageVersion("targets")`, and verify that the number you just output is greater than or equal to `0.3.1`. 
````
Opened in issue #20

padilla410 commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+You'll be revising files in this repository shortly. To follow our team's standard git workflow, you should first clone this training repository to your local machine so that you can make file changes and commits there. 
+
+Open a git bash shell (Windows) or a terminal window (Mac) and change (`cd`) into the directory you work in for projects in R (for me, this is `~/Documents/R`). There, clone the repository and set your working directory to the new project folder that was created:
```
I am going to push up a commit to do this, but should we be so explicit here? For example, different folks might have differing opinions on where to put stuff. For example, I organize by project so my directory would probably be something like `~/Documents/projects/training` for this one.

I'm on the fence about whether to do this:
```
cd Documents/R
git clone git@github.com:<username>/ds-pipelines-targets-1-course.git
cd ds-pipelines-targets-1-course
```

Or this:
```
cd <your preferred working directory>
git clone git@github.com:<username>/ds-pipelines-targets-1-course.git
cd ds-pipelines-targets-1-course
``

padilla410 replied:
I went with the latter

padilla410 commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+<hr> 
+
+<details>
+<summary><h2>Getting Started With USGS Data Science Pipelines</h2></summary>
```
Currently the doc is linked at line 212 [here](https://github.com/padilla410/ds-pipelines-targets-1-course-static/blob/main/course-instructions.md?plain=1#L212). I will also link at the top of the doc

padilla410 commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+Often we create temporary code or are sent scripts that look like `my_work_R/my_happy_script.R` in this repository. Take a minute to look through that file now. 
+
+This code has some major issues, including that it uses a directory that is specific to a user, it plots to a non-project file location, and the structure of the code makes it hard to figure out what is happening. This simple example is a starting point for understanding the investments we make to move towards code that is more reproducible, more shareable, and understandable. Additionally, we wa ...
```
I'm going to open this up as an issue - to be completed after this initial PR (#21)

padilla410 commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+### Background on phases 
+
+We use team conventions for how our pipelines are organized, which make it easier to hop in and out of collaborative projects and to rapidly understand what is going on where. 
```
Adding to #21

padilla410 commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+We refer to major elements of a pipeline as "phases", and name phases according to their purpose, such as `1_fetch` or `2_process`. These phases are used to separate files and data based on the _intent_ of the code we are writing, and make it tractable to figure out where you'd need to edit code if you were coming in fresh to the project.
+
+For medium to large pipelines projects, you'll see these workflow phases explicitly named by a number often followed by a verb (separated by an underscore). We use these phases to create different folders :file_folder: for data and code, and also to specify how we orchestrate the running of code (more on that later).
```
Yes, I like having numbered phases, but i typically do `01` and `02`, etc. I think it is also helpful to have numbers when we start using different names for phases (which is sometimes necessary)

padilla410 commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+So, if we have a `1_fetch` phase, code in the fetch folder :file_folder: would be used to do things like get data from web services, google drive, an FTP, or to scrape a website. `2_process` (or `2_munge`) might contain code that transforms the "fetched" data into more usable formats. 
+
+We recommend having `src` and `out` folders within each phase folder that contain code _for_ this phase (`src`) and data (or other files) produced _by_ this phase (`out`). When seeing some of our existing pipelines in action, you will also see other folders :file_folder: named `in`, `log`, and `tmp` to represent manually added files, logged/diagnostic output, and temporary data files, respectivel ...
+
```
#22

padilla410 commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+<hr> 
+
+We're going to combine several tasks into this assignment. You'll be asked to make some modifications to your training repository and also to create a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests) that captures these changes. 
```
I bumped it up into the intro and then add a lil transition text

padilla410 commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+Create a two phase directory structure for "fetch" and "process" concepts, and include `src` and `out` folders. Move the example script (`my_happy_script.R`) from the `my_work_R` folder into one of the `src` folders (at this time, it doesn't matter which one you choose) and delete any existing folders that aren't part of the intended structure.
+
+When you are done, open a `pull request` with the changes.
```
yes, I think so, right now that is documented in issue #18 for now

padilla410 commented on course-instructions.md:
```diff
@@ -0,0 +1,342 @@
+
+In addition to phases, it is important to decompose high-level concepts (or existing scripts) into thoughtful functions and _targets_ that form the building blocks of data processing pipelines. In addition to being the name of the pipeline package, a target is a noun we use to describe a tangible output of function, often a file or an R object, that we can use as an end-product (like a summary ma ...
+
+We strive to create _functions_ that declare their clear purpose (combining good function naming with thoughtful arguments/inputs is helpful for this) and are designed for re-use when appropriate. When writing pipelines functions, look for areas of re-usable operations or places where simple dividing lines can be drawn between different parts of data access, processing, modeling/analysis, and vis ...
```
Added to #21

padilla410 commented:
Ok! Cee, thanks for the review, I believe I have addressed all of your comments by either fixing them or by creating issues for them (I did this when a given comment did not seem like a quick fix).

I did this for all comments except for the one about linking the targets documentation at the top of the lesson. I did not make that change because I think I agree with the current flow of the lesson - start linking to the targets doc when you start drilling down into targets.
